observer/probers/crl/crl.go

```diff
@@ -47,6 +51,19 @@ func (p CRLProbe) Probe(timeout time.Duration) (bool, time.Duration) {
 		return false, dur
 	}
 
+	// Partitioned CRLs MUST contain an issuingDistributionPoint extension, which
+	// MUST contain the URL from which they were fetched, to prevent substitution
+	// attacks.
+	if p.partitioned {
+		idps, err := idp.GetIDPURIs(crl.Extensions)
+		if err != nil {
+			return false, dur
+		}
+		if len(idps) != 0 && !slices.Contains(idps, p.url) {
```

Comment:
The comment and the implementation are at odds. The comment indicates that a zero-length slice of IDPs should error, but the implementation will allow it.

Reply:
Oh good catch, the `if len(idps) != 0` is left over from before I realized that I needed to introduce the `partitioned` config parameter in order to do this correctly. I'll put up another PR to remove it.